src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/dto/ReferralDTOTest.kt

```diff
@@ -0,0 +1,73 @@
+package uk.gov.justice.digital.hmpps.hmppsinterventionsservice.dto
+
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Nested
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertThrows
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.util.ReferralFactory
+
+class ReferralDTOTest {
+  val referralFactory = ReferralFactory()
+
+  @Nested
+  inner class EndedReferral {
+    @Test
+    fun `endedReferralFields are populated when referral has endedAt`() {
+      val endedReferral = referralFactory.createEnded()
+      val referralDTO = ReferralDTO.from(endedReferral)
+      assertThat(referralDTO.endedFields).isNotNull()
+      assertThat(referralDTO.endedFields!!.endedAt).isEqualTo(endedReferral.endedAt)
```

Comment:
here you can use `endedFields?.endedAt` which returns either the field or null instead of making the not null assertion.

Reply:
see https://kotlinlang.org/docs/null-safety.html#safe-calls